File: frontend/src/lib/api/sessionSettings.ts
/**
 * Session settings API functions for voice input preferences
 */

import { ApiClient } from './client';

const api = new ApiClient();

/**
 * セッション設定（バックエンドレスポンス形式）
 */
export interface SessionSettingsResponse {
  session_id: string;
  voice_input_enabled: boolean;
  auto_send: boolean;
  mic_permission_checked: boolean;
}

/**
 * セッション設定（フロントエンド形式）
 */
export interface SessionSettings {
  voiceInputEnabled: boolean;
  autoSend: boolean;
  micPermissionChecked: boolean;
}

/**
 * セッション設定更新リクエスト
 */
interface SessionSettingsUpdateRequest {
  voice_input_enabled: boolean;
  auto_send: boolean;
  mic_permission_checked: boolean;
}

/**
 * セッションの音声入力設定を取得する
 *
 * @param sessionId - セッションID
 * @returns セッション設定（camelCase形式）
 *
 * @example
 * ```ts
 * const settings = await getSessionSettings('abc-123');
 * console.log(settings.voiceInputEnabled);
 * ```
 */
export async function getSessionSettings(
  sessionId: string
): Promise<SessionSettings> {
  const response = await api.get<SessionSettingsResponse>(
    `/api/v1/session-settings/${sessionId}`
  );

  // Convert snake_case to camelCase
  return {
    voiceInputEnabled: response.voice_input_enabled,
    autoSend: response.auto_send,
    micPermissionChecked: response.mic_permission_checked,
  };
}

/**
 * セッションの音声入力設定を更新する
 *
 * @param sessionId - セッションID
 * @param settings - 更新する設定（camelCase形式）
 * @returns 更新後の設定（camelCase形式）
 *
 * @example
 * ```ts
 * const updated = await updateSessionSettings('abc-123', {
 *   voiceInputEnabled: true,
 *   autoSend: true,
 *   micPermissionChecked: true,
 * });
 * ```
 */
export async function updateSessionSettings(
  sessionId: string,
  settings: SessionSettings
): Promise<SessionSettings> {
  const request: SessionSettingsUpdateRequest = {
    voice_input_enabled: settings.voiceInputEnabled,
    auto_send: settings.autoSend,
    mic_permission_checked: settings.micPermissionChecked,
  };

  const response = await api.put<SessionSettingsResponse>(
    `/api/v1/session-settings/${sessionId}`,
    request
  );

  return {
    voiceInputEnabled: response.voice_input_enabled,
    autoSend: response.auto_send,
    micPermissionChecked: response.mic_permission_checked,
  };
}

File: backend/app/api/v1/session_settings.py
"""Session settings API endpoints for voice input preferences"""

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.repositories.session_settings_repository import SessionSettingsRepository
from app.schemas.session_settings import SessionSettingsResponse, SessionSettingsUpdate

router = APIRouter()


@router.get("/{session_id}", response_model=SessionSettingsResponse)
def get_session_settings(
    session_id: str,
    db: Session = Depends(get_db),
) -> SessionSettingsResponse:
    """セッションの音声入力設定を取得する

    Args:
        session_id: セッションID

    Returns:
        セッション設定（存在しない場合はデフォルト値を返す）
    """
    repo = SessionSettingsRepository(db)
    settings = repo.get_by_session_id(session_id)

    if not settings:
        # Return default settings if not found
        return SessionSettingsResponse(
            session_id=session_id,
            voice_input_enabled=False,
            auto_send=True,
            mic_permission_checked=False,
        )

    return SessionSettingsResponse.model_validate(settings)


@router.put("/{session_id}", response_model=SessionSettingsResponse)
def update_session_settings(
    session_id: str,
    settings_update: SessionSettingsUpdate,
    db: Session = Depends(get_db),
) -> SessionSettingsResponse:
    """セッションの音声入力設定を更新する

    Args:
        session_id: セッションID
        settings_update: 更新する設定

    Returns:
        更新後のセッション設定
    """
    repo = SessionSettingsRepository(db)
    settings = repo.create_or_update(
        session_id=session_id,
        voice_input_enabled=settings_update.voice_input_enabled,
        auto_send=settings_update.auto_send,
        mic_permission_checked=settings_update.mic_permission_checked,
    )

    return SessionSettingsResponse.model_validate(settings)
